static/app/views/dashboards/widgetCard/releaseWidgetQueries.tsx

```diff
@@ -145,267 +136,256 @@ export function requiresCustomReleaseSorting(query: WidgetQuery): boolean {
+    !isEqual(
+      widget.queries.flatMap(q => q.fields?.filter(field => !!field)),
+      prevProps.widget.queries.flatMap(q => q.fields?.filter(field => !!field))
+    ) ||
+    !isEqual(
+      widget.queries.flatMap(q => q.aggregates.filter(aggregate => !!aggregate)),
+      prevProps.widget.queries.flatMap(q => q.aggregates.filter(aggregate => !!aggregate))
+    ) ||
+    !isEqual(
+      widget.queries.flatMap(q => q.columns.filter(column => !!column)),
+      prevProps.widget.queries.flatMap(q => q.columns.filter(column => !!column))
+    ) ||
+    loading !== prevProps.loading ||
+    cursor !== prevProps.cursor
+  );
+}
 
-  componentWillUnmount() {
-    this._isMounted = false;
-  }
+function ReleaseWidgetQueries({
+  widget,
+  selection,
+  dashboardFilters,
+  cursor,
+  limit,
+  onDataFetched,
+  onDataFetchStart,
+  children,
+}: ReleaseWidgetQueriesProps) {
+  const config = ReleasesConfig;
+
+  const mounted = useRef(false);
+  const allProjects = useProjects();
+  const api = useApi();
+  const organization = useOrganization();
+  const [requestErrorMessage, setRequestErrorMessage] = useState<string | undefined>(
+    undefined
+  );
+  const [releases, setReleases] = useState<Release[] | undefined>(undefined);
 
-  config = ReleasesConfig;
-  private _isMounted = false;
+  useEffect(() => {
+    mounted.current = true;
+    return () => {
+      mounted.current = false;
+    };
+  }, []);
 
-  fetchReleases = async () => {
-    this.setState({loading: true, errorMessage: undefined});
-    const {selection, api, organization, dashboardFilters} = this.props;
-    const {environments, projects} = selection;
+  const fetchReleases = useCallback(async () => {
+    setRequestErrorMessage(undefined);
 
     try {
-      const releases = await api.requestPromise(
+      const releaseResponse = await api.requestPromise(
         `/organizations/${organization.slug}/releases/`,
         {
           method: 'GET',
           data: {
             sort: 'date',
-            project: projects,
+            project: selection.projects,
             per_page: 50,
-            environment: environments,
+            environment: selection.environments,
             // Propagate release filters
             query: dashboardFilters
               ? dashboardFiltersToString(pick(dashboardFilters, 'release'))
               : undefined,
           },
         }
       );
-      if (!this._isMounted) {
+      if (!mounted.current) {
         return;
       }
-      this.setState({releases, loading: false});
+      setReleases(releaseResponse);
     } catch (error: any) {
-      if (!this._isMounted) {
+      if (!mounted.current) {
         return;
       }
 
       const message = error.responseJSON
         ? error.responseJSON.error
         : t('Error sorting by releases');
-      this.setState({errorMessage: message, loading: false});
+      setRequestErrorMessage(message);
```

Comment:
These are widget level request errors right, if not i'd recommend using our page level error provider
https://github.com/getsentry/sentry/blob/8365ec0174c1781dc1404457d9a788a599f5a62f/static/app/utils/performance/contexts/pageAlert.tsx#L69-L75